### PR TITLE
[Proposed] [fixes #16] ensure a missing FN in forEach doesn’t throw, rather retu…

### LINKED ISSUE #16: forEach returns a promise, but may also throw (if fn is not a function)
https://github.com/zenparsing/es-observable/blob/master/src/Observable.js#L194-L208

functions who can return a promise typically always return a promise.

### PR DESCRIPTION
**Pull request by stefanpenner**
…rns a rejected promise

Please note: 

* I am not that familiar with spec writing. So feel free to give feedback on the spec writing without

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -215,6 +214,7 @@ The *forEach* function performs the following steps:
 1. ReturnIfAbrupt(*observerNext*).
 1. Let *promiseCapability* be NewPromiseCapability(%Promise%).
 1. ReturnIfAbrupt(*promiseCapability*).
+1. If IsCallable(*callbackfn*) is **false**, Call(promiseCapability.[[Reject]], TypeError exception).
```
Comment (zenparsing) [inline README.md]: I think this line should be:

```
1. If Callable(*callbackfn*) is **false**,
    1. Let *rejectResult* be Call(*promiseCapability*.[[Reject]], **undefined**, «a newly created **TypeError** object»).
    1. ReturnIfAbrupt(*rejectResult*).
    1. Return *promiseCapability*.[[Promise]].
```

Comment (stefanpenner) [inline README.md]: isn't 

> 1. Return *promiseCapability*.[[Promise]].

implied by being in a promise constructor +  throwing ? Or am i confusing JS side-affects with spec lang requirements?

File: README.md [hunk 2]
```diff
@@ -215,6 +214,10 @@ The *forEach* function performs the following steps:
 1. ReturnIfAbrupt(*observerNext*).
 1. Let *promiseCapability* be NewPromiseCapability(%Promise%).
 1. ReturnIfAbrupt(*promiseCapability*).
+1. If Callable(*callbackfn*) is **false**,
```
Comment (zenparsing) [inline README.md]: Did I write "Callable"?  Oops : )  Should be `IsCallable`

Comment (stefanpenner) [inline README.md]: i should have re-read it. My bad for copying :(

**Pull request merged**